marklogic-data-hub-central/ui/src/components/load/load-card.test.tsx

```diff
@@ -1,10 +1,118 @@
 import React from 'react';
-import { shallow } from 'enzyme';
+import { render, fireEvent, wait, cleanup, waitForElement } from '@testing-library/react';
+import { MemoryRouter } from 'react-router-dom';
 import LoadCard from './load-card';
+import data from '../../config/test-data.config';
+import axiosMock from 'axios';
+import mocks from '../../config/mocks.config';
+import { AuthoritiesService, AuthoritiesContext } from '../../util/authorities';
+
+jest.mock('axios');
+
+const mockHistoryPush = jest.fn();
+
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useHistory: () => ({
+    push: mockHistoryPush,
+  }),
+}));
 
 describe('Load Card component', () => {
-  it('should render correctly', () => {
-    shallow(<LoadCard data deleteLoadArtifact
-      createLoadArtifact/>);
+
+  beforeEach(() => {
+    mocks.loadAPI(axiosMock);
+  })
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    cleanup();
+  })
+
+  test('Load Card - Add step to an existing Flow', async () => {
+    const authorityService = new AuthoritiesService();
+    authorityService.setAuthorities(['readIngestion', 'writeIngestion', 'writeFlow']);
+    const { getByText, getByLabelText, getByTestId } = render(
+      <MemoryRouter>
+        <AuthoritiesContext.Provider value={authorityService}>
+          <LoadCard 
+            {...data.loadData}
+            flows={data.flows}
+            canWriteFlow={true}
+            addStepToFlow={jest.fn()}
+            addStepToNew={jest.fn()} />
+        </AuthoritiesContext.Provider>
+      </MemoryRouter>
+    )
+
+    //Check if the card is rendered properly
+    expect(getByText('Add New')).toBeInTheDocument();
+    expect(getByText('testLoadXML')).toBeInTheDocument();
+    expect(getByLabelText('testLoadXML-sourceFormat')).toBeInTheDocument();
+    expect(getByText('Last Updated: 04/15/2020 2:22PM')).toBeInTheDocument();
+    expect(await (waitForElement(() => getByLabelText('testLoadXML-stepName')))).toBeInTheDocument();
+
+    fireEvent.mouseOver(getByText('testLoadXML')); // Hover over the Load Card to get more options
+
+    //Verify if the flow related options are availble on mouseOver
+    expect(getByTestId('testLoadXML-toNewFlow')).toBeInTheDocument(); // check if option 'Add to a new Flow' is visible
```

Comment:
At any point only one step could be moused over to display "Add to New/Existing Flow" right. Do we need testid. We can just use getByText.

Reply:
Options "Add to New/Existing Flow" are available for every card in the document, but except for the one we hover at any point, all others are marked as 'display: none'. So thats why i could't use just the text "Add to New/Existing Flow" and had to provide them a proper unique test id.